### ISSUE TITLE
Application panicked when starting prank with custom sender within `setUp`

### ISSUE THREAD

**Issue by jaycenhorton** [foundry-rs/foundry#5598]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (3fec8c1 2023-08-11T00:15:50.024503184Z)

_Note that this issue does not occur on versions <= nightly-d4f626bb7f96d46358997d4b27f79358cb2b3401_

### What command(s) is the bug in?

`forge test --sender=0x0000000000000000000000000000000000000001`

### Operating System

Linux

### Describe the bug

When running forge with a custom sender (either defined via `--sender` or `foundry.toml`), starting a prank in a `setUp` function causes the application to panick if the `test` function calls another address. I've created a [minimal repro repo here](https://github.com/jaycenhorton/forge-panick-repro). The following is the relevant test contract for convenience: 

```solidity
pragma solidity ^0.8.4;

import "forge-std/Test.sol";

contract Contract is Test {
  function setUp() external {
    vm.startPrank(address(1));
    // vm.stopPrank(); // uncomment this to fix
  }

  function test() external {
    // vm.stopPrank(); // OR uncomment this to fix
    payable(address(2)).transfer(1);
    // vm.stopPrank(); // uncommenting this will NOT fix
  }
}
```

To repro, run `forge test --sender=0x0000000000000000000000000000000000000001`. 

Here's the output

```
forge test --sender=0x0000000000000000000000000000000000000001                                    
[⠒] Compiling...
[⠘] Compiling 1 files with 0.8.17
[⠊] Solc 0.8.17 finished in 605.37ms
Compiler run successful!
The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: /home/runner/.cargo/git/checkouts/revm-bf744d8ffabcbad9/bc4d203/crates/revm/src/evm_impl.rs:267

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
[1]    88835 IOT instruction (core dumped)  forge test --sender=0x0000000000000000000000000000000000000001
```

For a more complex case, see [this PR](https://github.com/nori-dot-eco/contracts/pull/677) where it was initially encountered and worked around.

**Comment by mattsse** [comment]
thanks for the repro

@Evalir this looks like something's wrong with sender management in the evm's state, could also be a regression of the revm bump

**Comment by Evalir** [comment]
Closed with #5610
